test/helpers/setup-helper.js

```diff
@@ -89,10 +91,45 @@ window.document.body.appendChild(popoverContent);
 // eslint-disable-next-line no-shadow
 const { default: fetch, Headers, Request, Response } = require('node-fetch');
 
-Object.assign(window, { fetch, Headers, Request, Response });
+const handleRelativePathRequest = async (url, localeRelativePathRequest) => {
```

Comment:
Interesting solution! As an alternative to this, perhaps we can update `ui/contexts/i18n.js` to import the `en` localized messages rather than fetching them at runtime.  I see that we're already doing this in `shared/modules/i18n.js`. This would remove the need for shimming fetch I think.

Reply:
Ooooh this is interesting... We can definitely do that. But there's two things:
1. by importing it we are now including it in the bundled js. Currently, the `en` translations are included in the common.js bundle. They are also included as static assets in the `_locales` dir at the root (where the static.js build script is placing them). Probably the impact on the bundle size is neglectable  (en messages is about 220kb), but still we are duplicating it.
2. _locales are within the `app` directory. We probably want to move it from there, to avoid having the `ui` importing `app` files directly.
3. there's actually a bug on the ui code 😅 

So the [setupLocale](https://github.com/MetaMask/metamask-extension/blob/develop/ui/index.js#L88-L90) method that is run when initialising the UI to set the current locale messages (and the en fallback) in the redux store, always returns an empty object (the [fetchLocale](https://github.com/MetaMask/metamask-extension/blob/develop/shared/modules/i18n.ts#L89-L101) is wrapped on a try/catch that returns an empty object in case there's an error). This means if we fail to fetch/load the local files, the redux store will have this initial state:
```
    localeMessages: {
      ...
      current: {},
      en: {},
    },
```
The problem is that the [UI i18n context](https://github.com/MetaMask/metamask-extension/blob/develop/ui/contexts/i18n.js#L14-L22) and the [getMessage](https://github.com/MetaMask/metamask-extension/blob/develop/ui/helpers/utils/i18n-helper.tsx#L19) method does not handle well the empty object. The getMessage relies on the shared version of it which handles falsy localMessages but not an empty object. In that case it will throw a [missingKeyError](https://github.com/MetaMask/metamask-extension/blob/develop/shared/modules/i18n.ts#L63-L72)
.

Reply:
Also the [setupLocale](https://github.com/MetaMask/metamask-extension/blob/develop/ui/index.js#L88-L90) method seems buggy as the if clause to load the current locale relative time format and date will always be true - https://github.com/MetaMask/metamask-extension/blob/develop/shared/lib/error-utils.js#L13 🤔

Reply:
Update to the above.. the `missingKeyError` will only throw if we are running the app in test mode (process.env.IN_TEST). That is why the tests were failing. The Fallback seems to be applied if we are not running in test mode.
Still is there a reason to throw when running in test? 🤔

Reply:
hm...This is interesting. I'm not aware of any specific reason for the decision to throw missingKeyError only during testing 🤔, It might be worth discussing whether this behavior is necessary or should we reconsider the approach.